packages/php-wasm/compile/php/phpwasm-emscripten-library.js

```diff
@@ -270,6 +260,20 @@ const LibraryExample = {
 			return 0;
 		}
 
+		if (descriptorsLength < 2) {
```

Comment:
Oh no, I added this condition because I thought it had no sense to add only one descriptor. Sorry, subjective thought. What I know too is that when you add only one element in a dynamic array, the array has to add `16` to find the first element in memory. Let me give you an explanation about this : [ elements added in array -> pointer offset ] : 

1 -> 16
2 -> 16
3 -> 16
4 -> 24
5 -> 24
6 -> 32
7 -> 32
8 -> 48
9 -> 48
...

This means you can understand how to find the correct data in the memory even if it is dynamic. But there is a minimum of 16. And that `if( descriptorsLength < 2 )` allows to avoid 'if one element then + 16 instead of 8. 

I hope I explained this well.

Reply:
> Oh no, I added this condition because I thought it had no sense to add only one descriptor. Sorry, subjective thought.

All good! I just checked and it seems like you can even pass zero descriptors and `proc_open` would still work:

```php
$pipes = []; proc_open('echo "test123" > out', [], $pipes);
```